consensus/quorum/one-node-staked-vote.go

```diff
@@ -24,49 +28,88 @@ type TallyResult struct {
 	theirPercent numeric.Dec
 }
 
-type voteBox struct {
-	voters       map[shard.BLSPublicKey]struct{}
-	currentTotal numeric.Dec
+type tallyAndQuorum struct {
+	tally          numeric.Dec
+	quorumAchieved bool
 }
 
-type box struct {
-	Prepare    *voteBox
-	Commit     *voteBox
-	ViewChange *voteBox
+// VoteTally is the vote tally for each phase
+type VoteTally struct {
+	Prepare    *tallyAndQuorum
+	Commit     *tallyAndQuorum
+	ViewChange *tallyAndQuorum
 }
 
 type stakedVoteWeight struct {
 	SignatureReader
 	DependencyInjectionWriter
 	DependencyInjectionReader
 	roster    votepower.Roster
-	ballotBox box
+	voteTally VoteTally
 }
 
 // Policy ..
 func (v *stakedVoteWeight) Policy() Policy {
 	return SuperMajorityStake
 }
 
-// IsQuorumAchieved ..
-func (v *stakedVoteWeight) IsQuorumAchieved(p Phase) bool {
-	t := v.QuorumThreshold()
-	currentTotalPower, err := v.computeCurrentTotalPower(p)
+// AddNewVote ..
+func (v *stakedVoteWeight) AddNewVote(
+	p Phase, pubKeyBytes shard.BLSPublicKey,
+	sig *bls.Sign, headerHash common.Hash,
+	height, viewID uint64) (*votepower.Ballot, error) {
+
+	ballet, err := v.SubmitVote(p, pubKeyBytes, sig, headerHash, height, viewID)
 
 	if err != nil {
-		utils.Logger().Error().
-			AnErr("bls error", err).
-			Msg("Failure in attempt bls-key reading")
-		return false
+		return ballet, err
 	}
 
+	// Accumulate total voting power
+	additionalVotePower := v.roster.Voters[pubKeyBytes].OverallPercent
+	tallyQuorum := func() *tallyAndQuorum {
+		switch p {
+		case Prepare:
+			return v.voteTally.Prepare
+		case Commit:
+			return v.voteTally.Commit
+		case ViewChange:
+			return v.voteTally.ViewChange
+		default:
+			// Should not happen
+			return nil
+		}
+	}()
+
+	tallyQuorum.tally = tallyQuorum.tally.Add(additionalVotePower)
```

Comment:
do you need to continue adding `additionalVotePower` and recomputing `quorumAchieved` if it is already met? No need right?

Reply:
additionalVotePower accumulation is still needed, but the quorumCheck can be avoided. I will do it in next PR.